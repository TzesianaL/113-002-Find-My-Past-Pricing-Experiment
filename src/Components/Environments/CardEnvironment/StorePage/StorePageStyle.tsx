import styled from 'styled-components'
import { CardGroupStyle } from '../CardGroup/CardGroupStyle'

export const StorePageStyle = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  margin-bottom: 3em;
  background: #f2f2f2;

  h2,
  h1 {
    color: black;
  }

  & > section {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 1rem;
  }

  ${CardGroupStyle} {
    margin-top: 2em;
    margin-bottom: 2em;
    // width: 1540px;
  }

  .introText {
    font-size: 1.2em;
    max-width: 50em;
    @media screen and (max-width: 600px) {
      img {
        width: 100%;
      }
    }
  }

  .footerText {
    @media screen and (max-width: 600px) {
      max-width: 17em;
    }
    font-size: 1.2em;
    max-width: 50em;
  }
`

export const NewAddonGroupStyle = styled.div`
  display: flex;
  flex-direction: row !important;
  width: 100%;
  justify-content: center;
  flex-wrap: wrap;
`
