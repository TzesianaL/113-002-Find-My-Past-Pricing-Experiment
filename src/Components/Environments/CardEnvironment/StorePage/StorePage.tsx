import Button from 'Components/Shared/Button/Button'
import {
  MutexButton,
  MutexButtonGroup,
} from 'Components/Shared/Mutex/MutexButton'
import { useScrollToTop } from 'hooks/useScrollToTop'
import React, {
  FC,
  useEffect,
  useMemo,
  useRef,
  useState,
  useLayoutEffect,
} from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  getAllAddons,
  getSelectedCondition,
} from 'redux/reducers/dataStore/dataStore.selectors'
import {
  addItemToBasket,
  addSpeedItemToBasket,
} from 'redux/reducers/logicStore/logicStore'
import { getCost } from 'redux/reducers/logicStore/logicStore.selectors'
import { addOutput, OutputTypes } from 'redux/reducers/output/outputSlice'
import type { CardItem, SpeedCardItem } from 'types'
import { ContractDuration, isContractDuration } from 'types'
import { StorePageAddonLayout } from '../../../../StandardLayouts/AddonLayouts'
import StoreBanner from '../../../Shared/StoreBanner/StoreBanner'
import CardGroup from '../CardGroup/CardGroup'
import SpeedCardGroup from '../SpeedCardGroup/SpeedCardGroup'
import { StorePageStyle } from './StorePageStyle'

enum ConfigurationStep {
  CONTRACT_LENGTH = 'CONTRACT_LENGTH',
  SPEED_SELECTION = 'SPEED_SELECTION',
  PLAN_SELECTION = 'PLAN_SELECTION',
}

export interface StorePageProps {
  bannerName?: string
  bannerLogoSrc?: string
  introText?: string
  footerText?: string
  hidePrice?: boolean
  showCancelButton?: boolean
  purchaseButtonText?: string
  items: CardItem[]
  speedItems: SpeedCardItem[]
  onCancel: () => void
  onSelect: (id: string) => void
  onSpeedSelect?: (id: string) => void
  cancelButtonText?: string
}

const StorePage: FC<StorePageProps> = ({
  bannerLogoSrc,
  bannerName,
  introText,
  footerText,
  hidePrice,
  showCancelButton,
  purchaseButtonText,
  items,
  speedItems,
  onCancel,
  onSpeedSelect,
  onSelect,
  cancelButtonText,
}) => {
  const selectedCondition = useSelector(getSelectedCondition)
  const dispatch = useDispatch()

  const [contractLengthFilter, setContractLengthFilter] = useState<
    ContractDuration | undefined
  >(undefined)

  const [speedSelected, setSpeedSelected] = useState<string | undefined>(
    undefined
  )

  const [planSelected, setPlanSelected] = useState<string | undefined>(
    undefined
  )

  // Derive current step from state - this ensures consistency
  const currentStep = useMemo(() => {
    if (!contractLengthFilter) return ConfigurationStep.CONTRACT_LENGTH
    if (!speedSelected) return ConfigurationStep.SPEED_SELECTION
    return ConfigurationStep.PLAN_SELECTION
  }, [contractLengthFilter, speedSelected])

  // Scroll to relevant section when step changes
  useEffect(() => {
    const scrollToStep = () => {
      let targetRef: React.RefObject<HTMLDivElement> | null = null

      switch (currentStep) {
        case ConfigurationStep.CONTRACT_LENGTH:
          return
          break
        case ConfigurationStep.SPEED_SELECTION:
          targetRef = speedSelectionRef
          break
        case ConfigurationStep.PLAN_SELECTION:
          targetRef = planSelectionRef
          break
      }

      if (targetRef?.current) {
        targetRef.current.scrollIntoView({
          behavior: 'smooth',
          block: 'center',
          inline: 'nearest',
        })
      }
    }

    // Small delay to ensure DOM is updated before scrolling
    const timeoutId = setTimeout(scrollToStep, 100)
    return () => clearTimeout(timeoutId)
  }, [currentStep])

  const totalCost = useSelector(getCost)

  const handlePlanClick = (id: string) => {
    // Validation: ensure user can't skip steps
    if (!contractLengthFilter) {
      throw new Error('Cannot select plan without contract length')
    }
    setPlanSelected(id)
  }

  const handleSpeedSelect = (id: string) => {
    setSpeedSelected(id)
  }

  const handleContractLengthChange = (value: string | undefined) => {
    if (!value) {
      setContractLengthFilter(undefined)
      // Auto-reset dependent selections when contract changes
      setSpeedSelected(undefined)
      setPlanSelected(undefined)
      return
    }

    // Type-safe validation using type guard
    if (!isContractDuration(value)) {
      console.warn(`Invalid contract duration: ${value}`)
      return
    }

    setContractLengthFilter(value)
    // Auto-reset dependent selections when contract changes
    setSpeedSelected(undefined)
    setPlanSelected(undefined)
  }

  const addons = useSelector(getAllAddons)

  if (showCancelButton && !onCancel) {
    throw new Error(
      'OnCancel function is undefined when showCancelButton is provided'
    )
  }

  useEffect(() => {
    dispatch(
      addOutput({
        key: OutputTypes.itemOrderShown,
        value: items.map((item) => item.id).toString(),
      })
    )

    dispatch(
      addOutput({
        key: OutputTypes.numberOfItemsShown,
        value: String(items.length),
      })
    )
  }, [dispatch, items, speedItems])

  const durationData = useMemo(() => {
    const durationData = items.reduce(
      (acc, item) => {
        if (item.contractDuration) {
          if (!acc.durations.has(item.contractDuration)) {
            acc.durations.add(item.contractDuration)
            acc.cheapestPrices.set(item.contractDuration, item.price)
          } else {
            // Update if we found a cheaper price
            const currentCheapest = acc.cheapestPrices.get(
              item.contractDuration
            )!
            if (item.price < currentCheapest) {
              acc.cheapestPrices.set(item.contractDuration, item.price)
            }
          }
        }
        return acc
      },
      {
        durations: new Set<ContractDuration>(),
        cheapestPrices: new Map<ContractDuration, number>(),
      }
    )

    if (durationData.durations.size === 1) {
      const singleDuration = durationData.durations.values().next().value
      if (contractLengthFilter !== singleDuration) {
        handleContractLengthChange(singleDuration)
      }
    }

    return durationData
  }, [items, handleContractLengthChange])

  const DurationOptions: FC<{}> = () => {
    // Build unique durations and calculate cheapest prices in one pass
    const buttonOptions = Array.from(durationData.durations).map((duration) => {
      let buttonText: string
      const cheapestPrice = durationData.cheapestPrices.get(duration) || 0

      switch (duration) {
        case '24 months':
          buttonText = `24-month broadband, from £${cheapestPrice} a month`
          break
        case '30 days':
          buttonText = `30-day broadband, from £${cheapestPrice} per month`
          break
        default:
          buttonText = duration
          break
      }

      return (
        <MutexButton
          key={duration}
          value={duration}
          style={{ backgroundColor: 'white', borderRadius: '15px' }}
        >
          {buttonText}
        </MutexButton>
      )
    })

    return <>{buttonOptions}</>
  }

  const filteredSpeedItems = useMemo(() => {
    if (contractLengthFilter === undefined) {
      return []
    }

    const filteredSpeedIDs = items
      .filter((item) => item.contractDuration === contractLengthFilter)
      .reduce((acc, item) => {
        acc.add(item.parentSpeedId)
        return acc
      }, new Set<string>())

    return speedItems.filter((item) => filteredSpeedIDs.has(item.id))
  }, [items, speedItems, contractLengthFilter])

  const filteredItems = useMemo(() => {
    return items.filter(
      (item) =>
        item.contractDuration === contractLengthFilter &&
        item.parentSpeedId === speedSelected
    )
  }, [items, contractLengthFilter, speedSelected])

  useScrollToTop()

  // #region scroll to relevant section when step changes
  const speedSelectionRef = useRef<HTMLDivElement>(null)
  const contractLengthSelectionRef = useRef<HTMLDivElement>(null)
  const planSelectionRef = useRef<HTMLDivElement>(null)
  const isAutoSelecting = useRef(false)

  // Custom scroll function that uses center if element fits, otherwise start with offset
  const smartScrollIntoView = (
    element: HTMLElement,
    offsetFromTop: number = 80
  ) => {
    const elementRect = element.getBoundingClientRect()
    const viewportHeight = window.innerHeight
    const elementHeight = elementRect.height

    // Check if element would fit entirely on screen when centered
    const wouldFitWhenCentered = elementHeight <= viewportHeight * 0.8 // Use 80% of viewport for safety margin

    console.group('Smart Scroll Debug')
    if (wouldFitWhenCentered) {
      console.log(`Element fits, so scrolling to center`)
      // Use native center behavior if element fits comfortably
      element.scrollIntoView({
        behavior: 'smooth',
        block: 'center',
        inline: 'nearest',
      })
    } else {
      console.log(
        `Element does not fit when centered, scrolling with offset ${offsetFromTop}px`
      )
      // Calculate manual scroll position with offset
      const elementTop = element.offsetTop
      const targetScrollPosition = elementTop - offsetFromTop

      window.scrollTo({
        top: Math.max(0, targetScrollPosition), // Ensure we don't scroll above the page
        behavior: 'smooth',
      })
    }
    console.groupEnd()
  }

  // Scroll to relevant section when step changes
  useLayoutEffect(() => {
    let targetRef: React.RefObject<HTMLDivElement> | null = null

    switch (currentStep) {
      case ConfigurationStep.SPEED_SELECTION:
        // We have to handle this separately to ensure we scroll to speed selection but not on the intial autoselect
        return
      case ConfigurationStep.CONTRACT_LENGTH:
        targetRef = contractLengthSelectionRef
        break
      case ConfigurationStep.PLAN_SELECTION:
        targetRef = planSelectionRef
        break
    }

    if (targetRef?.current) {
      smartScrollIntoView(targetRef.current, 20) // 20px offset from top
    }
  }, [currentStep, speedSelected, contractLengthFilter])

  // Separate effect for scrolling to speed selection when broadband type is manually changed
  useLayoutEffect(() => {
    // Only scroll if broadband type was manually selected (not auto-selected)
    // and we're now at the speed selection step
    if (
      !isAutoSelecting.current &&
      currentStep === ConfigurationStep.SPEED_SELECTION &&
      speedSelectionRef.current
    ) {
      smartScrollIntoView(speedSelectionRef.current, 20)
    }

    // Reset the auto-selecting flag after processing
    if (isAutoSelecting.current) {
      isAutoSelecting.current = false
    }
  }, [currentStep, isAutoSelecting])
  // #endregion
  return (
    <StorePageStyle>
      <StoreBanner name={bannerName} logoSrc={bannerLogoSrc} />
      <h1>Build your broadband plan</h1>
      {introText && (
        <div
          className="introText"
          dangerouslySetInnerHTML={{ __html: introText }}
        />
      )}
      <>
        {/* Step 1: Contract Length Selection */}
        {durationData.durations.size > 1 && (
          <p style={{ fontSize: '1.25rem' }}>
            First, pick your <strong>contract length</strong>
          </p>
        )}
        {durationData.durations.size === 1 && (
          <p style={{ fontSize: '1.25rem', marginTop: '1.5em' }}>
            <strong>Contract length</strong>{' '}
          </p>
        )}

        <MutexButtonGroup
          value={contractLengthFilter}
          onValueChange={handleContractLengthChange}
        >
          <DurationOptions />
        </MutexButtonGroup>

        {/* Step 2: Speed Selection */}
        {(currentStep === ConfigurationStep.SPEED_SELECTION ||
          currentStep === ConfigurationStep.PLAN_SELECTION) && (
          <section ref={speedSelectionRef} style={{ marginTop: '2.5em' }}>
            <p style={{ fontSize: '1.25rem' }}>
              Select a <strong>speed</strong>
            </p>

            <SpeedCardGroup
              items={filteredSpeedItems.sort((a, b) => a.speed - b.speed)}
              hidePrice={hidePrice}
              selected={speedSelected}
              onSelect={onSpeedSelect}
              selectedCondition={selectedCondition}
              handleCardClick={handleSpeedSelect}
              purchaseButtonText={purchaseButtonText}
            />
          </section>
        )}

        {/* Step 3: Plan Selection */}
        {currentStep === ConfigurationStep.PLAN_SELECTION && (
          <section ref={planSelectionRef}>
            {selectedCondition !== 2 && (
              <p style={{ fontSize: '1.25rem', marginBottom: '0' }}>
                Now choose a plan with a <strong>{contractLengthFilter}</strong>{' '}
                contract
              </p>
            )}
            {selectedCondition === 2 && (
              <p style={{ fontSize: '1.25rem', marginBottom: '0' }}>
                Your <strong>{contractLengthFilter}</strong> contract include
              </p>
            )}

            <CardGroup
              items={filteredItems}
              hidePrice={hidePrice}
              selected={planSelected}
              onSelect={onSelect}
              handleCardClick={handlePlanClick}
              purchaseButtonText={purchaseButtonText}
              // avgScore={sumOfWeights}
            />
          </section>
        )}
      </>
      {footerText && (
        <div
          className="footerText"
          dangerouslySetInnerHTML={{ __html: footerText }}
        />
      )}
      <br />
      <br />
    </StorePageStyle>
  )
}

export default StorePage
