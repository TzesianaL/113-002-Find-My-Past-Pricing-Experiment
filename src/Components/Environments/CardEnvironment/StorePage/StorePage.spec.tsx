import React from 'react'
import 'jest-styled-components'
import configureStore from 'redux-mock-store'
import * as renderer from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { testItems } from 'stories/CardEnvironment/__fixtures__/card.fixtures'
import StorePage from './StorePage'

const mockStore = configureStore([])

describe('<StorePage/>', () => {
  let store

  beforeEach(() => {
    store = mockStore({
      testState: 'test txt',
    })
  })
  const TestStorePage = (props) => (
    <Provider store={store}>
      <StorePage
        items={testItems}
        onSelect={() => {}}
        bannerName="test banner-name"
        bannerLogoSrc="test banner-logo"
        {...props}
      />
    </Provider>
  )
  it('should render correctly StorePage', () => {
    expect(renderer.create(<TestStorePage />).toJSON()).toMatchSnapshot()
  })

  it("should render Cancel Button when showCancelButton is true and display default button text 'I Don't Want Any Of These Options'", () => {
    const showButton = true
    const button = render(
      <TestStorePage onCancel={() => {}} showCancelButton={showButton} />
    ).getByRole('button', { name: /I Don't Want Any Of These Options/i })
    expect(button).toBeInTheDocument()
  })
  it("should not render 'Cancel' Button when showCancelButton property is not provided ", () => {
    const button = render(<TestStorePage />)
    expect(button.queryByText(/I Don't Want Any Of These Options/)).toBeFalsy()
  })

  it('should throw an Error when showCancelButton is set to true and onCancel function is not provided', () => {
    const mockError = jest.spyOn(console, 'error')
    mockError.mockImplementation(() => {})
    expect(() => {
      render(<TestStorePage showCancelButton />)
    }).toThrowError(
      'OnCancel function is undefined when showCancelButton is provided'
    )

    mockError.mockRestore()
  })
  it('should call onCancel function,when showCancelButton is true and make sure it is clicked ', () => {
    const mockedFunction = jest.fn()
    render(<TestStorePage showCancelButton onCancel={mockedFunction} />)
    fireEvent.click(
      screen.getByRole('button', { name: /I Don't Want Any Of These Options/i })
    )
    expect(mockedFunction).toHaveBeenCalledTimes(1)
  })

  it('should show the provided cancelButtonText on the Cancel Button', () => {
    const testLabel = 'Cancel Button Text-Test'
    const mockedFunction = jest.fn()
    const el = render(
      <TestStorePage
        cancelButtonText={testLabel}
        showCancelButton
        onCancel={mockedFunction}
      />
    ).getByText(testLabel)
    expect(el).toBeInTheDocument()
  })

  it("should render the default value of I Don't Want Any Of These Options when cancelButtonText isn't  provided", () => {
    const mockedFunction = jest.fn()
    const button = render(
      <TestStorePage showCancelButton onCancel={mockedFunction} />
    ).getByRole('button', { name: /I Don't Want Any Of These Options/i })

    expect(button).toBeTruthy()
  })
})
