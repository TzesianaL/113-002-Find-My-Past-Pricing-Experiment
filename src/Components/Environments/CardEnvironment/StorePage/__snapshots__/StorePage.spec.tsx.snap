// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StorePage/> should render correctly StorePage 1`] = `
.c16 {
  color: #ffffff;
  background-color: #003D59;
  cursor: pointer;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0px;
  border-radius: 6px;
  padding: 0.7em 1em;
}

.c16:active {
  background-color: #336D89;
}

.c1 {
  background-color: #6300AA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fill-available;
  width: -moz-available;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white !important;
  padding: 0.6rem 0 0.6rem 0;
}

.c1 p {
  width: 100%;
  font-size: rem;
}

.c1 img {
  padding: 0 1em;
}

.c8 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c9 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #ffffff;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c11 {
  color: black;
  font-size: em;
  padding-left: 2em;
  padding-right: 2em;
  width: calc(100% - 4em);
}

.c11 li {
  padding-left: 0.25em;
  margin-bottom: 1.5em;
}

.c11 li::marker {
  content: '✓';
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1.2rem;
  margin: 0em 0em 0.5em 0em;
}

.c13 del {
  color: red;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 26%;
}

.c5 {
  text-align: center;
  background-color: #c8c2E8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-size: undefinedem;
  border: solid 2px;
  border-color: #773481;
  border-radius: 9px;
  width: 350px;
 
  color: black;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.12);
  margin: 0;
  margin: 0em 1em 1em 1em;
}

.c5 .c7 {
  color: black;
}

.c5 .c12 {
  color: black;
}

.c5 .c15 {
  width: 280px;
}

.c5 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 hr {
  width: 180px;
}

.c17 {
  text-align: center;
  background-color: #c2E8c8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-size: undefinedem;
  border: solid 2px;
  border-color: #773481;
  border-radius: 9px;
  width: 350px;
  height: 290px;
  color: black;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.12);
  margin: 0;
  margin: 0em 1em 1em 1em;
}

.c17 .c7 {
  color: black;
}

.c17 .c12 {
  color: black;
}

.c17 .c15 {
  width: 280px;
}

.c17 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c17 hr {
  width: 180px;
}

.c18 {
  text-align: center;
  background-color: #E8c2c8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-size: undefinedem;
  border: solid 2px;
  border-color: #773481;
  border-radius: 9px;
  width: 350px;
  height: 290px;
  color: black;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.12);
  margin: 0;
  margin: 0em 1em 1em 1em;
}

.c18 .c7 {
  color: black;
}

.c18 .c12 {
  color: black;
}

.c18 .c15 {
  width: 280px;
}

.c18 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c18 hr {
  width: 180px;
}

.c6 {
  padding: 16px 1em 24px 1em;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 1em;
  background: #333333;
}

.c3 .c4 {
  margin-left: 1em;
  margin-right: 1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 3em;
  background: #333333;
}

.c0 .c2 {
  margin-top: 2em;
  margin-bottom: 2em;
}

.c0 .introText {
  font-size: 1.2em;
  max-width: 50em;
}

.c0 .footerText {
  font-size: 1.2em;
  max-width: 50em;
}

@media screen and (max-width:600px) {
  .c5 {
    margin: 0 0 1em 0;
    height: auto;
    max-width: 100%;
    width: auto;
  }
}

@media screen and (max-width:600px) {
  .c17 {
    margin: 0 0 1em 0;
    height: auto;
    max-width: 100%;
    width: auto;
  }
}

@media screen and (max-width:600px) {
  .c18 {
    margin: 0 0 1em 0;
    height: auto;
    max-width: 100%;
    width: auto;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
    width: 100%;
  }

  .c3 .c4 {
    margin-left: 0;
    margin-right: 0;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="logo"
      height="44"
      src="test banner-logo"
    />
    <p>
      test banner-name
    </p>
  </div>
  <div
    className="c2 c3"
  >
    <div
      className="c4 c5"
      role="article"
    >
      <div
        className="c6"
      >
        <div
          className="c7 c8"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Basic",
              }
            }
          />
        </div>
        <div
          className="c7 c9"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "For beginners",
              }
            }
          />
        </div>
        <ul
          className="c10 c11"
        >
          Feature 1
        </ul>
        <hr />
        <div
          className="c12 c13"
        >
          £9.99
        </div>
        <div
          className="c14"
        >
          <button
            className="c15 c16"
            onClick={[Function]}
          >
            Choose Plan
          </button>
        </div>
      </div>
    </div>
    <div
      className="c4 c17"
      role="article"
    >
      <div
        className="c6"
      >
        <div
          className="c7 c8"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Better",
              }
            }
          />
        </div>
        <div
          className="c7 c9"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "For intermediates",
              }
            }
          />
        </div>
        <ul
          className="c10 c11"
        >
          Feature 1
        </ul>
        <hr />
        <div
          className="c12 c13"
        >
          £29.99
           per year
        </div>
        <div
          className="c14"
        >
          <button
            className="c15 c16"
            onClick={[Function]}
          >
            Choose Plan
          </button>
        </div>
      </div>
    </div>
    <div
      className="c4 c18"
      role="article"
    >
      <div
        className="c6"
      >
        <div
          className="c7 c8"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Premium",
              }
            }
          />
        </div>
        <div
          className="c7 c9"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "For professionals",
              }
            }
          />
        </div>
        <ul
          className="c10 c11"
        >
          Feature 1
        </ul>
        <hr />
        <div
          className="c12 c13"
        >
          £39.99
        </div>
        <div
          className="c14"
        >
          <button
            className="c15 c16"
            onClick={[Function]}
          >
            Choose Plan
          </button>
        </div>
      </div>
    </div>
  </div>
  <button
    className="c15 c16"
    onClick={[Function]}
  >
    I Don't Want Any Of These Options
  </button>
  <br />
  <br />
</div>
`;
